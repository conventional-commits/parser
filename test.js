const { describe, beforeEach, it } = require('mocha')
const { expect } = require('chai')
const chaiJestSnapshot = require('chai-jest-snapshot')
const parser = require('./')

require('chai')
  .use(chaiJestSnapshot)
  .should()

describe('<message>', () => {
  beforeEach(function () {
    chaiJestSnapshot.configureUsingMochaContext(this)
  })
  describe('<summary>', () => {
    it('parses summary with no scope', () => {
      const parsed = parser('fix: a really weird bug')
      parsed.should.matchSnapshot()
    })
    it('parses summary with scope', () => {
      const parsed = parser('feat(parser): add support for scopes')
      parsed.should.matchSnapshot()
    })
    it('parses breaking change marker', () => {
      let parsed = parser('feat!: add support for scopes')
      parsed.should.matchSnapshot()
      parsed = parser('feat(http parser)!: add support for scopes')
      parsed.should.matchSnapshot()
    })
    it('throws error when ":" token is missing', () => {
      expect(() => {
        parser('feat add support for scopes')
      }).to.throw("unexpected token ' ' at position 1:5 valid tokens [(, !, :]")
      expect(() => {
        parser('feat( foo ) add support for scopes')
      }).to.throw("unexpected token ' ' at position 1:12 valid tokens [!, :]")
      expect(() => {
        parser('feat(bar)! add support for breaking change')
      }).to.throw("unexpected token ' ' at position 1:11 valid tokens [:]")
    })
    it('throws error when closing ")" token is missing', () => {
      expect(() => {
        parser('feat(foo: add support for scopes')
      }).to.throw('unexpected token EOF valid tokens [)]')
    })
  })
  describe('<footer>', () => {
    it('parses simple token/separator/value form of footer', () => {
      const parsed = parser('fix: address major bug\nAuthor: @byCedric')
      parsed.should.matchSnapshot()
    })
    it('parses commit summary footer', () => {
      const parsed = parser('chore: contains multiple commits\nfix(parser): address bug with parser')
      parsed.should.matchSnapshot()
    })
    it('parses commit summary footer, with breaking change marker', () => {
      const parsed = parser('chore: contains multiple commits\nfeat(parser)!: address bug with parser')
      parsed.should.matchSnapshot()
    })
    it('parses BREAKING CHANGE literal as <token>', () => {
      const parsed = parser('fix: address major bug\nBREAKING CHANGE: this change is breaking')
      parsed.should.matchSnapshot()
    })
    it('supports multiline BREAKING CHANGES, via continuation', () => {
      const parsed = parser('fix: address major bug\nBREAKING CHANGE: first line of breaking change\n second line of breaking change\n third line of breaking change')
      parsed.should.matchSnapshot()
    })
  })
  describe('<summary>, <newline>*, <body>', () => {
    it('treats multiple newline between body and summary as optional', () => {
      const parsed = parser('fix: address major bug\nthis is a free form body of text')
      parsed.should.matchSnapshot()
    })
    it('allows for multiple newlines between summary and body', () => {
      const parsed = parser('fix: address major bug\n\nthis is a free form body of text')
      parsed.should.matchSnapshot()
    })
  })
  describe('[<text>], <newline>, <footer>*', () => {
    it('parses footer after body', () => {
      const parsed = parser('fix: address major bug\n\nthis is a free form body of text\nAuthor: @bcoe\nRefs #392')
      parsed.should.matchSnapshot()
    })
    it('parses footer after multi-line body', () => {
      const parsed = parser('fix: address major bug\n\nthis is the first line of the body\n\nthis is the second line of body\n\nAuthor: @bcoe\nRefs #392')
      parsed.should.matchSnapshot()
    })
  })
})
