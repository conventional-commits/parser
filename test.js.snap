// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<message> <footer> parses BREAKING CHANGE literal as <token> 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "fix",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "start": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 23,
              "line": 1,
              "offset": 22,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "text",
          "value": "address major bug",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 23,
          "line": 1,
          "offset": 22,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 16,
                      "line": 2,
                      "offset": 38,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 2,
                      "offset": 23,
                    },
                  },
                  "type": "breaking-change",
                  "value": "BREAKING CHANGE",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 16,
                  "line": 2,
                  "offset": 38,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 23,
                },
              },
              "type": "token",
            },
            Object {
              "position": Object {
                "end": Object {
                  "column": 18,
                  "line": 2,
                  "offset": 40,
                },
                "start": Object {
                  "column": 16,
                  "line": 2,
                  "offset": 38,
                },
              },
              "type": "separator",
              "value": ": ",
            },
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 41,
                      "line": 2,
                      "offset": 63,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 2,
                      "offset": 40,
                    },
                  },
                  "type": "text",
                  "value": "this change is breaking",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 41,
                  "line": 2,
                  "offset": 63,
                },
                "start": Object {
                  "column": 18,
                  "line": 2,
                  "offset": 40,
                },
              },
              "type": "value",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 41,
              "line": 2,
              "offset": 63,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 23,
            },
          },
          "type": "footer",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 41,
          "line": 2,
          "offset": 63,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 23,
        },
      },
      "type": "body-footer",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 41,
      "line": 2,
      "offset": 63,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <footer> parses commit summary footer 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "chore",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 33,
              "line": 1,
              "offset": 32,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "text",
          "value": "contains multiple commits",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 33,
          "line": 1,
          "offset": 32,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 4,
                      "line": 2,
                      "offset": 36,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 2,
                      "offset": 33,
                    },
                  },
                  "type": "type",
                  "value": "fix",
                },
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 11,
                      "line": 2,
                      "offset": 43,
                    },
                    "start": Object {
                      "column": 5,
                      "line": 2,
                      "offset": 37,
                    },
                  },
                  "type": "scope",
                  "value": "parser",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 12,
                  "line": 2,
                  "offset": 44,
                },
                "start": Object {
                  "column": 37,
                  "line": 2,
                  "offset": 69,
                },
              },
              "type": "token",
            },
            Object {
              "position": Object {
                "end": Object {
                  "column": 14,
                  "line": 2,
                  "offset": 46,
                },
                "start": Object {
                  "column": 12,
                  "line": 2,
                  "offset": 44,
                },
              },
              "type": "separator",
              "value": ": ",
            },
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 37,
                      "line": 2,
                      "offset": 69,
                    },
                    "start": Object {
                      "column": 14,
                      "line": 2,
                      "offset": 46,
                    },
                  },
                  "type": "text",
                  "value": "address bug with parser",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 37,
                  "line": 2,
                  "offset": 69,
                },
                "start": Object {
                  "column": 14,
                  "line": 2,
                  "offset": 46,
                },
              },
              "type": "value",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 37,
              "line": 2,
              "offset": 69,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 33,
            },
          },
          "type": "footer",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 37,
          "line": 2,
          "offset": 69,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 33,
        },
      },
      "type": "body-footer",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 37,
      "line": 2,
      "offset": 69,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <footer> parses simple token/separator/value form of footer 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "fix",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "start": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 23,
              "line": 1,
              "offset": 22,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "text",
          "value": "address major bug",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 23,
          "line": 1,
          "offset": 22,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 7,
                      "line": 2,
                      "offset": 29,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 2,
                      "offset": 23,
                    },
                  },
                  "type": "type",
                  "value": "Author",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 7,
                  "line": 2,
                  "offset": 29,
                },
                "start": Object {
                  "column": 18,
                  "line": 2,
                  "offset": 40,
                },
              },
              "type": "token",
            },
            Object {
              "position": Object {
                "end": Object {
                  "column": 9,
                  "line": 2,
                  "offset": 31,
                },
                "start": Object {
                  "column": 7,
                  "line": 2,
                  "offset": 29,
                },
              },
              "type": "separator",
              "value": ": ",
            },
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 18,
                      "line": 2,
                      "offset": 40,
                    },
                    "start": Object {
                      "column": 9,
                      "line": 2,
                      "offset": 31,
                    },
                  },
                  "type": "text",
                  "value": "@byCedric",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 18,
                  "line": 2,
                  "offset": 40,
                },
                "start": Object {
                  "column": 9,
                  "line": 2,
                  "offset": 31,
                },
              },
              "type": "value",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 18,
              "line": 2,
              "offset": 40,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 23,
            },
          },
          "type": "footer",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 18,
          "line": 2,
          "offset": 40,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 23,
        },
      },
      "type": "body-footer",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 18,
      "line": 2,
      "offset": 40,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <footer> supports multiline BREAKING CHANGES, via continuation 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "fix",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "start": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 23,
              "line": 1,
              "offset": 22,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "text",
          "value": "address major bug",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 23,
          "line": 1,
          "offset": 22,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 16,
                      "line": 2,
                      "offset": 38,
                    },
                    "start": Object {
                      "column": 1,
                      "line": 2,
                      "offset": 23,
                    },
                  },
                  "type": "breaking-change",
                  "value": "BREAKING CHANGE",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 16,
                  "line": 2,
                  "offset": 38,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 23,
                },
              },
              "type": "token",
            },
            Object {
              "position": Object {
                "end": Object {
                  "column": 18,
                  "line": 2,
                  "offset": 40,
                },
                "start": Object {
                  "column": 16,
                  "line": 2,
                  "offset": 38,
                },
              },
              "type": "separator",
              "value": ": ",
            },
            Object {
              "children": Array [
                Object {
                  "position": Object {
                    "end": Object {
                      "column": 47,
                      "line": 2,
                      "offset": 69,
                    },
                    "start": Object {
                      "column": 18,
                      "line": 2,
                      "offset": 40,
                    },
                  },
                  "type": "text",
                  "value": "first line of breaking change",
                },
                Object {
                  "children": Array [
                    Object {
                      "position": Object {
                        "end": Object {
                          "column": 32,
                          "line": 3,
                          "offset": 101,
                        },
                        "start": Object {
                          "column": 2,
                          "line": 3,
                          "offset": 71,
                        },
                      },
                      "type": "text",
                      "value": "second line of breaking change",
                    },
                  ],
                  "position": Object {
                    "end": Object {
                      "column": 32,
                      "line": 3,
                      "offset": 101,
                    },
                    "start": Object {
                      "column": 47,
                      "line": 2,
                      "offset": 69,
                    },
                  },
                  "type": "continuation",
                },
                Object {
                  "children": Array [
                    Object {
                      "position": Object {
                        "end": Object {
                          "column": 31,
                          "line": 4,
                          "offset": 132,
                        },
                        "start": Object {
                          "column": 2,
                          "line": 4,
                          "offset": 103,
                        },
                      },
                      "type": "text",
                      "value": "third line of breaking change",
                    },
                  ],
                  "position": Object {
                    "end": Object {
                      "column": 31,
                      "line": 4,
                      "offset": 132,
                    },
                    "start": Object {
                      "column": 32,
                      "line": 3,
                      "offset": 101,
                    },
                  },
                  "type": "continuation",
                },
              ],
              "position": Object {
                "end": Object {
                  "column": 31,
                  "line": 4,
                  "offset": 132,
                },
                "start": Object {
                  "column": 18,
                  "line": 2,
                  "offset": 40,
                },
              },
              "type": "value",
            },
          ],
          "position": Object {
            "end": Object {
              "column": 31,
              "line": 4,
              "offset": 132,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 23,
            },
          },
          "type": "footer",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 31,
          "line": 4,
          "offset": 132,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 23,
        },
      },
      "type": "body-footer",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 31,
      "line": 4,
      "offset": 132,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <summary> parses breaking change marker 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "feat",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "start": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
          },
          "type": "breaking-change",
          "value": "!",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 30,
              "line": 1,
              "offset": 29,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "text",
          "value": "add support for scopes",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 30,
          "line": 1,
          "offset": 29,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 30,
      "line": 1,
      "offset": 29,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <summary> parses breaking change marker 2`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "feat",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 17,
              "line": 1,
              "offset": 16,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "scope",
          "value": "http parser",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 19,
              "line": 1,
              "offset": 18,
            },
            "start": Object {
              "column": 18,
              "line": 1,
              "offset": 17,
            },
          },
          "type": "breaking-change",
          "value": "!",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 21,
              "line": 1,
              "offset": 20,
            },
            "start": Object {
              "column": 19,
              "line": 1,
              "offset": 18,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 43,
              "line": 1,
              "offset": 42,
            },
            "start": Object {
              "column": 21,
              "line": 1,
              "offset": 20,
            },
          },
          "type": "text",
          "value": "add support for scopes",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 43,
          "line": 1,
          "offset": 42,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 43,
      "line": 1,
      "offset": 42,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <summary> parses summary with no scope 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "fix",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
            "start": Object {
              "column": 4,
              "line": 1,
              "offset": 3,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 24,
              "line": 1,
              "offset": 23,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "text",
          "value": "a really weird bug",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 24,
          "line": 1,
          "offset": 23,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 24,
      "line": 1,
      "offset": 23,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;

exports[`<message> <summary> parses summary with scope 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 5,
              "line": 1,
              "offset": 4,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "type",
          "value": "feat",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 12,
              "line": 1,
              "offset": 11,
            },
            "start": Object {
              "column": 6,
              "line": 1,
              "offset": 5,
            },
          },
          "type": "scope",
          "value": "parser",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 15,
              "line": 1,
              "offset": 14,
            },
            "start": Object {
              "column": 13,
              "line": 1,
              "offset": 12,
            },
          },
          "type": "separator",
          "value": ": ",
        },
        Object {
          "position": Object {
            "end": Object {
              "column": 37,
              "line": 1,
              "offset": 36,
            },
            "start": Object {
              "column": 15,
              "line": 1,
              "offset": 14,
            },
          },
          "type": "text",
          "value": "add support for scopes",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 37,
          "line": 1,
          "offset": 36,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "summary",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 37,
      "line": 1,
      "offset": 36,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "message",
}
`;
